qa/L0_request_cancellation/client_cancellation_test.py

```diff
@@ -0,0 +1,246 @@
+                raise data_item
+        self.assertIn("Locally cancelled by application!", str(cm.exception))
+
+        self._record_end_time_ms()
+        self._test_runtime_duration(5000)
+
+    def test_grpc_stream_infer(self):
+        # Sends a request using async_stream_infer to a
+        # model that takes 10s to execute. Issues stream
+        # closure with cancel_requests=True. The client
+        # should return with appropriate exception within
+        # 5s.
+        triton_client = grpcclient.InferenceServerClient(
+            url="localhost:8001", verbose=True
+        )
+
+        self._prepare_request()
+        user_data = UserData()
+
+        triton_client.start_stream(callback=partial(callback, user_data))
+        self._record_start_time_ms()
+
+        with self.assertRaises(InferenceServerException) as cm:
+            for i in range(1):
+                triton_client.async_stream_infer(
+                    model_name=self.model_name_,
+                    inputs=self.inputs_,
+                    outputs=self.outputs_,
+                )
+                time.sleep(2)
+            triton_client.stop_stream(cancel_requests=True)
+            data_item = user_data._completed_requests.get()
+            if type(data_item) == InferenceServerException:
+                raise data_item
+        self.assertIn("Locally cancelled by application!", str(cm.exception))
+
+        self._record_end_time_ms()
+        self._test_runtime_duration(5000)
+
+    def test_aio_grpc_async_infer(self):
+        # Sends a request using infer of grpc.aio to a
+        # model that takes 10s to execute. Issues
+        # a cancellation request after 2s. The client
+        # should return with appropriate exception within
+        # 5s.
+        async def cancel_request(call):
+            await asyncio.sleep(2)
+            self.assertTrue(call.cancel())
+
+        async def handle_response(generator):
+            with self.assertRaises(asyncio.exceptions.CancelledError) as cm:
+                _ = await anext(generator)
+
+        async def test_aio_infer(self):
+            triton_client = aiogrpcclient.InferenceServerClient(
+                url="localhost:8001", verbose=True
+            )
+            self._prepare_request()
+            self._record_start_time_ms()
+
+            generator = triton_client.infer(
+                model_name=self.model_name_,
+                inputs=self.inputs_,
+                outputs=self.outputs_,
+                get_call_obj=True,
+            )
+            grpc_call = await anext(generator)
+
+            tasks = []
+            tasks.append(asyncio.create_task(handle_response(generator)))
+            tasks.append(asyncio.create_task(cancel_request(grpc_call)))
+
+            for task in tasks:
+                await task
+
+            self._record_end_time_ms()
+            self._test_runtime_duration(5000)
+
+        asyncio.run(test_aio_infer(self))
```

Comment:
I'm fine with we have but you could also use the `unittest.IsolatedAsyncioTestCase` for unit testing async methods. You can use this as a reference: 

https://github.com/triton-inference-server/server/blob/7624490df45549a2aa4f2ed1d5c963414a265c62/qa/L0_grpc/python_grpc_aio_test.py#L34C25-L34C57

Reply:
I have deferred adding asyncio request cancellation to a later point. We can explore better ways to implement and test asyncio library